warp/Network/Wai/Handler/Warp/Response.hs

```diff
@@ -351,6 +285,76 @@ sendRsp conn _ _ _ _ (RspRaw withApp src tickle) = do
     send bs = connSendAll conn bs >> tickle
 
 ----------------------------------------------------------------
+
+-- Sophisticated WAI applications.
+-- We respect s0. s0 MUST be a proper value of either 200 or 206.
```

Comment:
Ah. 404 is also OK. For instance, a HTML file for 404 is specified. This should be read "s0 MUST be a proper value". I will fix.